chore(github): SCRUM-44 Update GitHub Pull Request Template

adnlv opened the pull request:
**Jira**: [SCRUM-44](https://epam-team-php.atlassian.net/browse/SCRUM-44)

## Description

Standardize the process of creating pull requests and ensure that all pull requests include essential information.

## How to roll back?

<!-- How to roll back these changes -->

Revert this pull request.

## How has this been tested?

<!-- Describe the tests that were run to verify your changes -->
<!-- Provide reproducible instructions -->

## Media attachments (optional)

<!-- Drag & drop images here or paste links -->

adnlv committed:
chore(github): SCRUM-44 Simplify pull request template

There may be several PR templates, so it is more convenient to move
them to their own directory.

ihpr commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,21 @@
+<!-- Jira ticket URL in the format [SCRUM-X](https://.../SCRUM-X) -->
+**Jira**:
```
```suggestion
**Jira**: [SCRUM-X](https://epam-team-php.atlassian.net/browse/SCRUM-X)
```

ihpr commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,21 @@
+**Jira**:
+
+<!-- list of related issues -->
+Closes:
```
@adanyl0v How often is this thing gonna be used? Does it make sense to keep it under the template?

ihpr commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,21 @@
+
+<!-- Short summary of what this PR does -->
+
+## Test coverage
```
@adanyl0v According to the best practices, test coverage must be evaluated automatically.
Test coverage is a metric that defines which production code is covered by tests and which is not.

What do you mean by test coverage here?
Does it make sense to keep it in the PR template?
How often is it gonna be used?

ihpr reviewed (changes requested):
This template looked unfinsihed to me. I would suggest to consider next topics:
- Rollback plan (if comsthing goes wrong how the PR will be rollbacked does it require any additional rollback steps instead of a simple revert)
- Checklist (you can create some checklist based best practises applied by your team, e.g. commit messages are accurate, code has no SQL injections, code has proepr validation, code follow naming conventions, etc.)

Ideas:
<img width="715" height="735" alt="image" src="https://github.com/user-attachments/assets/0381a830-7fcb-46e7-9af5-73ca774fb07a" />

- Maybe it is worth considering "How to Test" section instead of "Test coverage"?

Please let me know does it make sense?

adnlv committed:
chore(github): SCRUM-44 Add Jira ticket link example

adnlv committed:
chore(github): SCRUM-44 Remove redundant closing issues section

adnlv committed:
chore(github): SCRUM-44 Improve pull request template clarity

Rename "Test coverage" to "How has this been tested?" for clearer
guidance and add a "How to roll back?" section to document reversion
steps.

ihpr reviewed (approved):
LGTM. For now it's ok. In the future it can be extended with checklist, etc.
